arcade/arcade/core/schema.py

```diff
@@ -324,10 +324,18 @@ def _get_item(
 
         normalized_key = key.lower()
         for item in items:
-            if item.key.lower() == normalized_key:
+            if item.key.lower() == normalized_key or item.key.upper() == normalized_key:
                 return item.value
+
         raise ValueError(f"{item_name.capitalize()} {key} not found in context.")
 
+    def set_secret(self, key: str, value: str) -> None:
+        """Set a secret for the tool invocation."""
+        if not self.secrets:
+            self.secrets = []
+        secret = ToolSecretItem(key=str(key).lower(), value=str(value))
```

Comment:
the `key` should remain the same casing as what the environment variable is. i.e., remove the `.lower()`.

The current behavior is equivalent to the following:
```
>export ABC=123
>echo $ABC // cannot find the envvar because it was converted to lower case
>echo $abc // echoes back 123
```